index.js

```diff
@@ -469,6 +476,28 @@ Router.prototype.use = function use(handler) {
     throw new TypeError('argument handler is required')
   }
 
+  var name
+
+  // If a name is used, the last argument will be a string, not a function
+  if (callbacks.length > 1 && typeof callbacks[callbacks.length - 1] !== 'function') {
```

Comment:
This is a little too liberal, due to the flatten above. For example, this would allow for `router.use([[[[function(){}]],'name']])` and pick up the name, when it probably needs to be restricted to being a top-level argument.